### DISCUSSION TITLE
Question on SchmidFactor

### DISCUSSION THREAD

**Discussion by bjzhou96** [mtex-toolbox/mtex#1311] [Ask Anything] [answered]
Hello everyone!

I am now working with Mtex and trying to plotting SchmidFactor map by it. I want to plot the SchmidFactor map for (100)<001>slip system of ampihole with a stess direction parallel with Z-axis direction. Here is my code:
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd);
ebsd(grains(grains.grainSize<10)) = [];
 [grains,ebsd.grainId] = calcGrains(ebsd,'angle','10*degree');
grains = smooth(grains,5);

% plot(ebsd);
% hold on
% plot(grains('Amphibole').boundary,'linewidth',1)
% hold off
cs = crystalSymmetry('12/m1', [9.9 18 5.4],[90,104.81,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Amphibole')
b = Miller(0,0,1,cs,'uvw')
n = Miller(1,0,0,cs,'hkl')
sS1 = slipSystem(b,n)
% rotate slip systems into specimen coordinates
sSLocal = grains('Amphibole').meanOrientation * sS1
% compute Schmid factor
sigma = stressTensor.uniaxial(vector3d.Z)
SF = sSLocal.SchmidFactor(sigma);

% take the maxium allong the rows
[SFMax,active] = max(SF,[],2);

% plot the maximum Schmid factor
plot(grains('Amphibole'),SFMax,'micronbar','off','linewidth',2)
mtexColorbar location southoutside
```
It works well and I have the SchimidFactor map :
![image](https://user-images.githubusercontent.com/58097480/147049806-c4d2bc62-6456-4a2e-ac1f-138c19543bf2.png)

However, in my knowledge, the Schmidfactor ranges 0-0.5. And 0 indicates that a crystal is in a difficult orientation for activation of a particular slip system while factors closer to∼0.5 indicate an easy orientation and therefore a more favorable slip system. However, as you can see, my Schmidfactor map ranges -0.5-0.5. 

So my question is (1) why the Schmidfactor map ranges -0.5-0.5 and how can we set it between 0-0.5? (2) how to set the stressTensor for a stress with a direction of 45° with X axis in XZ plane?

Thank you very much in advance!

Baojun Zhou

**Comment by kilir** [comment, marked as answer]
Hi,
the sign changes according to the direction of slip with respect to the slip system. E.g.
```matlab
cs = crystalSymmetry('1')
o = orientation.byEuler([0,0,0]*degree,cs)

h1= Miller(0,1,0,cs)
h2 = Miller(1,0,0,cs,'uvw')

sS = slipSystem(h2,h1)

plot(SchmidFactor(o.*sS))
hold on
plot(o*h1,'circle')
hold on
plot(o*h2,'label','b')
hold off
```
If this is not relevant for you, you can simply use the absolute value.

Cheers,
Rüdiger
